### DISCUSSION TITLE
classes/class_signal

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#176] [User-contributed notes]
# classes/class_signal

A built-in type representing a signal of an Object. Description: Signal is a built-in Variant type that represents a signal of an Object instance. Like all Variant types, it can be stored in variab...

https://docs.godotengine.org/en/stable/classes/class_signal.html

**Comment by timpni** [comment]
Why is the function has_signal( Signal_Name : String) not in here?

**Reply by Calinou** [reply]
`has_signal()` is part of Object, which is documented here: https://docs.godotengine.org/en/stable/classes/class_object.html#class-object-method-has-signal

Signal does **not** inherit from Object, so you cannot call `has_signal()` on a Signal.

Nonetheless, I've opened a PR to point to `Object.has_signal()` in Signal's description: https://github.com/godotengine/godot/pull/97734

**Reply by timpni** [reply]
Thanks! Yeah, a page regarding Signals should include all function used with signals, otherwise it becomes hard to find certain functionally. The reason i found has_signal() was due to chat GPT which i would not have needed to ask for help if it was here in the first place, but props for the fast response!

Same goes for the group page, I find it jarring having to sift through the page to find different functions.

**Comment by JamesDepret** [comment]
The C# example code is methods are missing the obligated "public" text

**Reply by Calinou** [reply]
Which code sample are you referring to, and where should the `public` keyword go?

I suggest opening a pull request on the main Godot repository by editing `Signal.xml` directly: https://github.com/godotengine/godot/blob/master/doc/classes/Signal.xml

**Comment by elementbound** [comment]
@Calinou are there any guarantees on call order? E.g. if I connect callables A, B and C, will they be called in the same order on emit?

I've heard some people say it's not guaranteed, but reading the source code, my understanding is that Signals store their connected Callables in HashMaps, which maintain order of insertion. So, in theory, when iterating through the callables in `emit`, it should run callables in the same order they were connected.

Edit: I've also failed to find official docs on this, but it's entirely possible I've just missed something :sweat_smile:

**Comment by Login1990** [comment]
It seems not to be written anywhere, but if you if you store `await`ed signal in a variable, it apparently returns all the arguments of a signal.

So `var x = await area2d.body_entered` creates reference to the body that has entered, found it by trial and error.

**Comment by konasquared** [comment]
The `connect` method description was a little misleading for me:
> A signal can only be connected once to the same Callable.

This led me to believe that a signal could have only one connection at all. I'm not one to read the entire documentation page, so while this is obvious if you read the entire document top-to-bottom, it's not if you, for instance, ctrl-click in the editor to see the `connect` description.

Can this be reworded slightly differently? Maybe like this: "Signals cannot be connected multiple times to the same Callable". I think it's a little less vague than the current wording.

**Comment by admin** [comment]
*Originally posted by [**ValorZard**](https://github.com/ValorZard) on 2026-06-16 21:49:26 UTC*

> Are signals thread safe?
